src/Magento/FunctionalTestingFramework/Console/StaticChecksCommand.php

```diff
@@ -164,4 +160,43 @@ private function validateInput(InputInterface $input)
                 );
         }
     }
+
+    /**
+     * Validates that all passed in static-check names match an existing static check
+     * @param string[] $requiredChecksNames
+     * @return void
+     */
+    private function validateTestNames($requiredChecksNames)
+    {
+        $invalidCheckNames = [];
+        for ($index = 0; $index < count($requiredChecksNames); $index++) {
+            if (in_array($requiredChecksNames[$index], array_keys($this->allStaticCheckObjects))) {
+                $this->staticCheckObjects[$requiredChecksNames[$index]] =
+                    $this->allStaticCheckObjects[$requiredChecksNames[$index]];
+            } else {
+                $invalidCheckNames[] = $requiredChecksNames[$index];
+            }
+        }
+
+        if (!empty($invalidCheckNames)) {
+            throw new InvalidArgumentException(
+                'Invalid static check script(s): ' . implode(', ', $invalidCheckNames) . '.'
+            );
+        }
+    }
+
+    /**
+     * Parses and sets local ruleSet. If not found, simply returns and lets script continue.
+     * @return void;
+     */
+    private function parseRulesetJson()
+    {
+        $pathToRuleset = TESTS_BP . DIRECTORY_SEPARATOR . "staticRuleset.json";
+        if ($pathToRuleset === null) {
+            $this->ioStyle->text("No ruleset under $pathToRuleset" . PHP_EOL);
+            return;
+        }
+        $this->ioStyle->text("Using ruleset under $pathToRuleset" . PHP_EOL);
+        $this->ruleSet = json_decode(file_get_contents($pathToRuleset), true);
```

Comment:
the step throws a fatal error if file is not found. Good to have exception handled.

Reply:
Fair point, will fix.